lib/active_fedora/noid/synchronized_minter.rb

```diff
@@ -1,35 +1,19 @@
 require 'noid'
+
 module ActiveFedora
```

Comment:
The DB minter is a new class, @mjgiarlo.  I refactored so that we can use common base class between the two.  I consider that the most certain and least objectionable portion of this work.  I deferred on renaming the existing `SynchronizedMinter` for back-compat reasons, but if you prefer, I certainly would move it to be something like `ActiveFedora::Noid::Minter::Statefile`.  

The syncedness (of threads in a single system, single process) is part of Base.  It does not need to be called out in naming (especially if the name suggests that it is the same problem as multiple processes on multiple systems using potentially multiple connections to the same database).  

If you want to give me a thumbs up, I'll pursue renaming the existing minter.  It may make sense to defer that to after getting this initial PR merged.